inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/user/TenantRoleServiceImpl.java

```diff
@@ -121,4 +122,17 @@ public TenantRoleInfo getByUsernameAndTenant(String name, String tenant) {
         return CommonBeanUtils.copyProperties(entity, TenantRoleInfo::new);
     }
 
+    @Override
+    public List<String> listTenantByUsername(String username) {
+        TenantRolePageRequest request = new TenantRolePageRequest();
+        request.setPageNum(1);
+        request.setPageSize(Integer.MAX_VALUE);
+        request.setUsername(username);
+        PageResult<TenantRoleInfo> tenantRoleInfoList = listByCondition(request);
+        return tenantRoleInfoList.getList().stream()
+                .map(TenantRoleInfo::getTenant)
+                .distinct()
```

Comment:
Why not do distinct in the SQL?

Reply:
In order to reuse **_listByCondition_**.
Usually, each user has only one role in each tenant.
The **_distinct_** process is to remove duplicate tenant if a user is both the ADMIN and OPERATOR in the same tenant.